plugins/metas.ts

```diff
@@ -71,8 +72,14 @@ export default function (userOptions?: Partial<Options>) {
         return;
       }
 
-      if (!metas.title && options.usePageTitle && page.data.title) {
-        metas.title = page.data.title as string;
+      if (options.defaultPageData) {
+        for (const [pageKey, k] of Object.entries(options.defaultPageData)) {
+          const metaKey = k as keyof MetaData;
+
+          if (!metas[metaKey] && page.data[pageKey]) {
+            updateDefaultMeta(metas, metaKey, page.data[pageKey]);
```

Comment:
Why do you need this function, instead of simply do `metas[metaKey] = page.data[pageKey] as MetaData[T];`

Reply:
Another problem I see here is the `metas` object is modified by this function. I would rather don't modify the page data to avoid possible edge cases. For example, something like this would have the same result, without modify the page data:

```js
function getValue(value, defaultKey, data) {
  if (value) {
    return value;
  }
  if (defaultKey) {
    return data[defaultKey];
  }
}
addMeta(document, "property", "og:title", getValue(metas.title, options.defaultPageData.title, page.data), 65);
```

Reply:
> Why do you need this function, instead of simply do `metas[metaKey] = page.data[pageKey] as MetaData[T];`

I need the `T` type params to make typescript happy as the above code dont have T speciifed.

Reply:
> Another problem I see here is the `metas` object is modified by this function. I would rather don't modify the page data to avoid possible edge cases. For example, something like this would have the same result, without modify the page data:
> 
> ```js
> function getValue(value, defaultKey, data) {
>   if (value) {
>     return value;
>   }
>   if (defaultKey) {
>     return data[defaultKey];
>   }
> }
> addMeta(document, "property", "og:title", getValue(metas.title, options.defaultPageData.title, page.data), 65);
> ```

You are right, I will work on this.

Reply:
Ok, could you do something that returns the value but without modify the object?
Similar to the function in the example of my previous comment.